libraries/chain/include/eosio/chain/block.hpp

```diff
@@ -109,9 +109,16 @@ namespace eosio { namespace chain {
     * what would be logged to disk to enable the regeneration of blockchain state.
     *
     * The transactions are grouped to mirror the cycles in block_summary, generated
-    * transactions are not included.  
+    * transactions are not included.
     */
    struct signed_block : public signed_block_summary {
+      signed_block () = default;
+      signed_block (const signed_block& ) = default;
+      signed_block (const signed_block_summary&& base)
+         :signed_block_summary (base),
+          input_transactions()
+      {}
```

Comment:
This is not moving `base`, it is copying it. You need `signed_block_summary(std::move(base))`.  Also it should not take the rvalue via const since you will not be able to move it. Also const rvalue reference is generally used to prevent binding to an rvalue which it would do if you also provided a non-const rvalue reference. You should also provide a default move constructor to prevent slicing when actually moving a signed_block.